### PR TITLE
Add Authentication without SSO article to KB

### PR DESCRIPTION
**Pull request by snehankekre**
- Adds a KB article on  `Authentication without SSO` to the Deployment issues section.
- Includes the following warning in the KB article:
    > While these techniques were tested and secured on the Streamlit side, they are obviously not comparable to proper authentication with a SSO provider. Please view them as a temporary workaround! For enterprise use-cases, please read [Configuring Single Sign-on (SSO)](https://docs.streamlit.io/streamlit-cloud/get-started/share-your-app/configuring-single-on-sso).
- Links to the KB article in the Troubleshooting section of Streamlit Cloud.

### CODE REVIEW

File: content/kb/deployments/authentication-without-sso.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
+---
+title: Authentication without SSO
+slug: /knowledge-base/deploy/authentication-without-sso
+---
+
+# Authentication without SSO
+
+## Introduction
+
+Want to secure your Streamlit app with passwords but Streamlit Cloud Teams and Enterprise doesn't support your SSO provider yet? We got you covered! This guide shows you two simple techniques for adding basic authentication to your Streamlit app, using [secrets management](/streamlit-cloud/get-started/deploy-an-app/connect-to-data-sources/secrets-management).
```
Comment (randyzwitch) [inline content/kb/deployments/authentication-without-sso.md:10]: By moving this to the knowledge base, we should probably revise this sentence to something like "Want to secure your Streamlit app with passwords, but cannot implement single sign-on?" This better reflects that community tier doesn't have the ability to do so, and Streamlit Cloud potentially doesn't support a specific provider

File: content/kb/deployments/authentication-without-sso.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+<Warning>
+
+While these techniques were tested and secured on the Streamlit side, they are obviously not comparable to proper authentication with a SSO provider. Please view them as a temporary workaround! For enterprise use-cases, please read [Configuring Single Sign-on (SSO)](/streamlit-cloud/get-started/share-your-app/configuring-single-on-sso).
```
Comment (randyzwitch) [inline content/kb/deployments/authentication-without-sso.md:14]: I'd remove the "While these techniques were tested and secured on the Streamlit side" for something more generic. Something to the effect of "While this technique adds some level of security, it is NOT comparable to proper authentication..."  I'd also remove "Please view them" sentence.

File: content/streamlit-cloud/troubleshooting.md [hunk 3]
```diff
@@ -81,9 +81,9 @@ We are working on this in Q4 of 2021. Check back soon for more information!
 
 If you are on the same GitHub repo then you will automatically be added to the same workspace. Just invite them to log in to [share.streamlit.io](http://share.streamlit.io) and we will automatically route them to your workspace once they hook in their GitHub account.
 
-<!-- ### How do I add viewers to my Streamlit apps?
+### How do I add viewers to my Streamlit apps?
 
-Before viewer authentication is added via single sign-on, [apps are secured via password protection](https://www.notion.so/Authentication-without-SSO-4319c6135b4b4ed58355fc06b33cac74). To give someone access, you just need to set up a password for them. -->
+Before viewer authentication is added via [single sign-on](/streamlit-cloud/get-started/share-your-app/configuring-single-on-sso), [apps are secured via password protection](/knowledge-base/deploy/authentication-without-sso). To give someone access, you just need to set up a password for them.
```
Comment (randyzwitch) [inline content/streamlit-cloud/troubleshooting.md:86]: We should change this sentence. The way it reads now, it seems to both imply that SSO isn't available and that apps are automatically secured via password protection. Neither part is true.

**Review by randyzwitch — APPROVED** [review]
Approved, pending revisions

File: content/kb/deployments/authentication-without-sso.md (hunk 1 above)
Comment (snehankekre) [reply content/kb/deployments/authentication-without-sso.md:10]: Agreed! I've changed it to "Want to secure your Streamlit app with passwords, but cannot implement single sign-on?"

File: content/streamlit-cloud/troubleshooting.md (hunk 3 above)
Comment (snehankekre) [reply content/streamlit-cloud/troubleshooting.md:86]: Good catch. It now reads:

> By default, all apps deployed with Streamlit Cloud Teams and Enterprise are private—which means that others in your company won't be able to view them unless you give them explicit permission. To add viewers, [configure single sign-on](/streamlit-cloud/get-started/share-your-app/configuring-single-on-sso) with your organization's SSO provider. 

> If you cannot implement single sign-on, but want to secure your Streamlit app with passwords, read our guide on [authentication without SSO](/knowledge-base/deploy/authentication-without-sso). Note: while this technique adds some level of security, it is **NOT** comparable to proper authentication with a SSO provider.

WDYT?

File: content/kb/deployments/authentication-without-sso.md (hunk 2 above)
Comment (snehankekre) [reply content/kb/deployments/authentication-without-sso.md:14]: Thank you! I've edited it to read:

> While this technique adds some level of security, it is **NOT** comparable to proper authentication with a SSO provider. For enterprise use-cases, please read [Configuring Single Sign-on (SSO)](/streamlit-cloud/get-started/share-your-app/configuring-single-on-sso).

**Comment by randyzwitch** [comment]
Thanks for the changes Snehan, looks great!

**Pull request merged**